src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/commands.py

```diff
@@ -3,41 +3,48 @@
                               get_monitor_log_profiles_operation,
                               get_monitor_autoscale_settings_operation,
                               get_monitor_diagnostic_settings_operation,
-                              get_monitor_activity_log_operation,
-                              get_monitor_metric_definitions_operation,
-                              get_monitor_metrics_operation)
+                              get_monitor_activity_log_operation)
+from azure.cli.core.commands import cli_command
+from azure.cli.core.commands.arm import cli_generic_update_command
 from azure.cli.core.sdk.util import (ServiceGroup, create_service_adapter)
 
 
 # MANAGEMENT COMMANDS
+
+custom_path = 'azure.cli.command_modules.monitor.custom#'
 custom_operations = create_service_adapter('azure.cli.command_modules.monitor.custom')
 
-alert_rules_operations = create_service_adapter(
-    'azure.mgmt.monitor.operations.alert_rules_operations',
-    'AlertRulesOperations')
+# region Alerts
 
-with ServiceGroup(__name__, get_monitor_alert_rules_operation, alert_rules_operations) as s:
-    with s.group('monitor alert-rules') as c:
-        c.command('create', 'create_or_update')
-        c.command('delete', 'delete')
-        c.command('show', 'get')
-        c.command('list', 'list_by_resource_group')
-        c.generic_update_command('update', 'get', 'create_or_update')
+ar_path = 'azure.mgmt.monitor.operations.alert_rules_operations#AlertRulesOperations.'
 
-alert_rule_incidents_operations = create_service_adapter(
-    'azure.mgmt.monitor.operations.alert_rule_incidents_operations',
-    'AlertRuleIncidentsOperations')
+cli_command(__name__, 'monitor alert create', custom_path + 'create_metric_rule', cf_alert_rules)
+cli_command(__name__, 'monitor alert delete', ar_path + 'delete', cf_alert_rules)
+cli_command(__name__, 'monitor alert show', ar_path + 'get', cf_alert_rules)
+cli_command(__name__, 'monitor alert list', ar_path + 'list_by_resource_group', cf_alert_rules)
+cli_generic_update_command(__name__, 'monitor alert update',
+                           ar_path + 'get',
+                           ar_path + 'create_or_update',
+                           cf_alert_rules,
+                           custom_function_op=custom_path + 'update_metric_rule')
 
-with ServiceGroup(__name__, get_monitor_alert_rule_incidents_operation,
-                  alert_rule_incidents_operations) as s:
-    with s.group('monitor alert-rule-incidents') as c:
-        c.command('show', 'get')
-        c.command('list', 'list_by_alert_rule')
+ari_path = 'azure.mgmt.monitor.operations.alert_rule_incidents_operations#AlertRuleIncidentsOperations.'
+cli_command(__name__, 'monitor alert show-incident', ari_path + 'get', cf_monitor)
+cli_command(__name__, 'monitor alert list-incidents', ari_path + 'list_by_alert_rule', cf_monitor)
+
+# endregion
+
+# region Metrics
+
+cli_command(__name__, 'monitor metrics list', custom_path + 'list_metrics', cf_metrics)
```

Comment:
I know that you are directly using `cli_command` instead of the pattern of `ServiceGroup` to command registration. It'd be good if we can keep this module to one pattern either one :)